building/glass/aptbranch.py

```diff
@@ -73,3 +42,41 @@ def export_key(keyid, armor=False):
     if not result.strip():
         raise Exception("empty result from gpg for keyid: '%s'" % keyid)
     return result
+
+
+def select_branch_config(branches_config: list, branch: str):
+    for config in branches_config:
+        if config["name"] == branch:
+            return config
+    return None
+
+
+class Config:
+    def __init__(self, branch: str):
+        if not os.path.exists(CONFIG_PATH):
+            raise Exception("cannot find branches config at %s, use %s.example to create one" % (CONFIG_PATH, CONFIG_PATH))
+        branches_config = validate.load_validated(CONFIG_PATH, CONFIG_SCHEMA_NAME)["branches"]
+        self.config = select_branch_config(branches_config, branch)
+
+        if self.config is None:
+            raise Exception("no config found for %s in %s" % (branch, CONFIG_PATH))
```

Comment:
Maybe it makes more sense to just directly throw this, rather than returning None and then checking for None?